cmd/titus-metadata-service/main.go

```diff
@@ -235,29 +202,44 @@ func main() {
 			EnvVar:      "X_FORWARDED_FOR_BLOCKING_MODE",
 			Destination: &xFordwardedForBlockingMode,
 		},
+		cli.StringFlag{
+			Name:        "peer-namespace",
+			Usage:       "When set, the proxy will bind inside the namespace specified",
+			EnvVar:      "PEER_NAMESPACE",
+			Destination: &peerNs,
+		},
 	}
+
 	app.Action = func(c *cli.Context) error {
 		if debug {
 			log.SetLevel(log.DebugLevel)
 		} else {
 			log.SetLevel(log.InfoLevel)
 		}
-		// This needs to be if journald available, because titus executors are started by mesos agent, and in order
-		// for logs to get from the tasks starts by titus tasks to
-		logsutil.MaybeSetupLoggerIfOnJournaldAvailable()
+
+		utils.MaybeSetupLoggerIfOnJournaldAvailable()
+
 		/* Get the requisite configuration from environment variables */
-		listener := getListener(listenPort, listenerFd)
+		var listener net.Listener
+
+		if len(peerNs) > 0 {
+			// We were launched by a CNI. Bind inside peer namespace.
+			log.Infof("Launched with PEER_NS %s, LISTEN_PORT %d", peerNs, listenPort)
+			nsListener, err := utils.GetNsListener(peerNs, listenPort)
+			if err != nil {
+				log.Fatalf("Error getting listener %s", err)
+			}
+
+			listener = nsListener
+		} else {
+			listener = getListener(listenPort, listenerFd)
```

Comment:
I think if you want to continue this behaviour (logic of getting the listener in this code), then should pass it into getListener, so getListener can validate that listenerPort XOR listener FD are set, and if listenerNS is set, then ensure listener FD is not set.

Reply:
We do _not_ pass an FD if we are running under k8s. Which is the reason for this structure.